### PR TITLE
Add x-ms-format mappings for DataFactoryExpression<T>

### PR DESCRIPTION
**Pull request by JoshLove-msft**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/AutoRest.CSharp.csproj [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
 
   <ItemGroup>
-    <PackageReference Include="Azure.Core" Version="1.25.0" />
+    <PackageReference Include="Azure.Core" Version="1.26.0" />
```
Comment (m-nash) [inline src/AutoRest.CSharp/AutoRest.CSharp.csproj:21]: In order for us to test this version bump we need to update [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpProj.cs#L36) and regen the entire library by running `./eng/Generate.ps1` from the root of the repo.

File: src/AutoRest.CSharp/Common/AutoRest/Communication/StandaloneGeneratorRunner.cs [hunk 2]
```diff
@@ -61,7 +62,7 @@ public static async Task RunAsync(CommandLineOptions options)
                 if (options.IsNewProject)
                 {
                     var codeModel = await codeModelTask;
-                    new CSharpProj().Execute(Configuration.Namespace ?? codeModel.Language.Default.Name, outputPath);
+                    new CSharpProj().Execute(Configuration.Namespace ?? codeModel.Language.Default.Name, outputPath, (yaml.Contains("x-ms-format: dfe-")));
```
Comment (m-nash) [inline src/AutoRest.CSharp/Common/AutoRest/Communication/StandaloneGeneratorRunner.cs:65]: add Culture.Ordinal for perf improvement?

File: samples/Azure.ResourceManager.Sample/Generated/Models/VirtualMachineScaleSetListOSUpgradeHistory.Serialization.cs [hunk 3]
```diff
@@ -38,7 +38,7 @@ internal static VirtualMachineScaleSetListOSUpgradeHistory DeserializeVirtualMac
                         property.ThrowNonNullablePropertyIsNull();
                         continue;
                     }
-                    etag = new ETag(property.Value.GetString());
+                    etag = JsonSerializer.Deserialize<ETag>(property.Value.ToString());
```
Comment (m-nash) [inline samples/Azure.ResourceManager.Sample/Generated/Models/VirtualMachineScaleSetListOSUpgradeHistory.Serialization.cs:41]: revert this for now and add an issue to track whether this is an overall safe / performant change

File: src/AutoRest.CSharp/Mgmt/Decorator/CodeModelTransformer.cs [hunk 4]
```diff
@@ -2,11 +2,13 @@
 // Licensed under the MIT License.
 
 using System.Collections.Generic;
+using System.Linq;
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/Decorator/CodeModelTransformer.cs:5]: nit: usings can probably go away

File: src/AutoRest.CSharp/Mgmt/Decorator/Transformer/JsonConverterResolver.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                {
+                    property.Extensions!["x-ms-format-element-type"] = schema;
+                    schema.Extensions ??= new RecordOfStringAndAny();
+                    if (!schema.Extensions!.ContainsKey("x-csharp-usage"))
+                    {
+                        schema.Extensions.Add("x-csharp-usage", "converter");
+                    }
+                    else
+                    {
+                        schema.Extensions!["x-csharp-usage"] += ",converter";
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/Decorator/Transformer/JsonConverterResolver.cs:40]: is there ever a case where it has this with an empty string?

Comment (JoshLove-msft) [reply src/AutoRest.CSharp/Mgmt/Decorator/Transformer/JsonConverterResolver.cs:40]: Updated to handle empty string.

**Comment by JoshLove-msft** [comment]
Created https://github.com/Azure/autorest.csharp/issues/2871 and https://github.com/Azure/autorest.csharp/issues/2870 as follow ups.

**Comment by JoshLove-msft** [comment]
> I've generated against azure-sdk-for-net with these changes and confirmed there is no diff.

This is no longer the case with the GetRawText change.

File: src/AutoRest.CSharp/Common/Decorator/SchemaUsageTransformer.cs [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+            }
+
+            if (schemaToPropertyMap.Count > 0)
+            {
+                var schemaList = schemaToPropertyMap.Keys.Aggregate((a, b) => $"{a}, {b}");
+                throw new InvalidOperationException($"The following schemas were referenced by properties with the '{FormatElementDefinition}' attribute, but were not found in any definitions: " + schemaList);
+            }
+        }
+
+        private static void Apply(ObjectSchema schema, HashSet<string> usages, HashSet<KnownMediaType> mediaTypes, HashSet<ObjectSchema> appliedSchemas)
```
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply src/AutoRest.CSharp/Common/Decorator/SchemaUsageTransformer.cs:112]: Yes, good catch.

**Comment by JoshLove-msft** [comment]
Diff - https://github.com/Azure/azure-sdk-for-net/pull/32822

**Review by m-nash — APPROVED** [review]

**Pull request merged**